src/Uno.UI.RuntimeTests/Helpers/TimeZoneModifier.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Reflection;
+
+namespace Uno.UI.RuntimeTests.Helpers;
+
+public class TimeZoneModifier : IDisposable
+{
+	private readonly TimeZoneInfo _originalTimeZoneInfo;
+
+	public TimeZoneModifier(TimeZoneInfo timeZoneInfo)
+	{
+		_originalTimeZoneInfo = GetLocalTimeZone();
+		SetLocalTimeZone(timeZoneInfo);
+	}
+
+	private static TimeZoneInfo GetLocalTimeZone()
+	{
+		var cachedData = typeof(TimeZoneInfo).GetField("s_cachedData", BindingFlags.Static | BindingFlags.NonPublic).GetValue(null);
```

Comment:
No problem with this, though we should make it explicit with an error when GetField returns null.